rust/lance-linalg/src/kmeans.rs

```diff
@@ -499,12 +489,7 @@ where
                                 .collect();
                         }
                         MetricType::Cosine => {
-                            let normalized = values
-                                .chunks(dimension)
-                                .flat_map(normalize)
-                                .collect::<Vec<_>>();
-                            return compute_partitions_l2(centroids_array, &normalized, dimension)
-                                .collect();
+                            panic!("KMeans: should not use cosine distance to train kmeans, use L2 instead.");
```

Comment:
add this check in the constructor?

Reply:
The match arm needs a branch to handle `MetricType::Cosine` tho.